src/walletProvider.ts

```diff
@@ -1,110 +1,74 @@
+import { ITransaction } from "./interface";
 import {
     WALLET_PROVIDER_CALLBACK_PARAM,
     WALLET_PROVIDER_CALLBACK_PARAM_TX_SIGNED,
     WALLET_PROVIDER_CONNECT_URL,
     WALLET_PROVIDER_DISCONNECT_URL,
-    WALLET_PROVIDER_SEND_TRANSACTION_URL,
     WALLET_PROVIDER_SIGN_TRANSACTION_URL,
 } from "./constants";
-import { ErrInvalidTxSignReturnValue, ErrNotImplemented } from "./errors";
-
-interface TransactionMessage {
-    receiver: string;
-    value: string;
-    gasPrice?: number;
-    gasLimit?: number;
-    data?: string;
-    nonce?: number;
-    options?: number;
-}
+import { ErrInvalidTxSignReturnValue } from "./errors";
+import { PlainSignedTransaction } from "./plainSignedTransaction";
 
 export class WalletProvider {
     private readonly walletUrl: string;
-    private readonly transactionFactory: ITransactionFactory;
 
     /**
      * Creates a new WalletProvider
      * @param walletURL
      */
-    constructor(walletURL: string, transactionFactory: ITransactionFactory) {
+    constructor(walletURL: string) {
         this.walletUrl = walletURL;
-        this.transactionFactory = transactionFactory;
     }
 
     /**
      * Fetches the login hook url and redirects the client to the wallet login.
      */
     async login(options?: { callbackUrl?: string; token?: string }): Promise<string> {
-        let callbackUrl = `callbackUrl=${window.location.href}`;
-        if (options && options.callbackUrl) {
-            callbackUrl = `callbackUrl=${options.callbackUrl}`;
-        }
+        let callbackUrl = options?.callbackUrl || window.location.href;
+        let redirectUrl = `${this.baseWalletUrl()}${WALLET_PROVIDER_CONNECT_URL}?callbackUrl=${callbackUrl}`;
 
-        let token = '';
-        if (options && options.token) {
-            token = `&token=${options.token}`;
+        if (options?.token) {
+            redirectUrl = `${redirectUrl}&token=${options.token}`
         }
 
-        const redirect = `${this.baseWalletUrl()}${WALLET_PROVIDER_CONNECT_URL}?${callbackUrl}${token}`;
-        // TODO: Perhaps do not delay, just like for signTransactions()?
         await new Promise((resolve) => {
             setTimeout(() => {
-              window.location.href = redirect;
-              resolve(true);
+                window.location.href = redirectUrl;
+                resolve(true);
             }, 10);
-          });
+        });
 
-        // TODO: Perhaps do not return anything?
-        return window.location.href;
+        return redirectUrl;
     }
 
     /**
     * Fetches the logout hook url and redirects the client to the wallet logout.
     */
     async logout(options?: { callbackUrl?: string }): Promise<boolean> {
-        let callbackUrl = `callbackUrl=${window.location.href}`;
-        if (options && options.callbackUrl) {
-            callbackUrl = `callbackUrl=${options.callbackUrl}`;
-        }
+        let callbackUrl = options?.callbackUrl || window.location.href;
```

Comment:
This is very similar to the lines in the login function, maybe these can be turned into "getRedirectUrl(options)"

Reply:
Refactored URL building.